src/app/calendar-app/runbox-calendar-event.ts

```diff
@@ -69,137 +88,437 @@ export class RunboxCalendarEvent implements CalendarEvent {
         this.event.description = value;
     }
 
+    // This is the start time of an event instance (if recurring)
+    // See also: calendar.service.ts generateEvents
     get dtstart(): moment.Moment {
-        return this.icalTimeToLocalMoment(this.event.startDate);
+        return this.icalTimeToLocalMoment(this._dtstart);
     }
 
+    // If set on an event with RRULE, updates entire repeated event
+    // to now *start* from this new date
+    // If set on an exception, only changes that exception
     set dtstart(value: moment.Moment) {
         const allDay = this.allDay;
-        this.event.startDate = ICAL.Time.fromJSDate(value.toDate());
+        this._dtstart = this.momentToIcalTime(value, this.event.startDate ? this.event.startDate.zone : null);
+        // FIXME: Do this only if its the first one?
+        this.event.startDate = this._dtstart;
+        // FIXME: regenerate all following events!?
+        // Do we just let the save-to-dav / reloadEvents cycle do this for us?
         this.allDay = allDay;
     }
 
     get dtend(): moment.Moment {
-        return this.event.endDate
-            ? this.icalTimeToLocalMoment(this.event.endDate)
+        return this._dtend
+            ? this.icalTimeToLocalMoment(this._dtend)
             : undefined;
     }
 
+    // If set on an event with RRULE, updates entire repeated event
+    // ?? Does this shorten it to one day ? (does it include the date)
+    // If set on an exception, only changes that exception
+    // FIXME: Does this change an ICAL.Time with no date (isDate=true) to
```

Comment:
Good question – I guess so, since Moment doesn't really have such a distinction, and it's impossible to infer what the user meant from the datetime alone. We may need to spread ICAL.Time further around our code to fix this :|

Reply:
We have tests using both dates and datetimes, so hopefully we're good..